src/keybindings.c

```diff
@@ -172,12 +173,47 @@ GeanyKeyBinding *keybindings_set_item(GeanyKeyGroup *group, gsize key_id,
 	kb->default_key = key;
 	kb->default_mods = mod;
 	kb->callback = callback;
+	kb->handler = NULL;
+	kb->handler_data = NULL;
 	kb->menu_item = menu_item;
 	kb->id = key_id;
 	return kb;
 }
 
 
+/** Creates a new keybinding using a GeanyKeyHandler and attaches is to a keybinding group
+ *
+ * If given the callback should return @c TRUE if the keybinding was handled, otherwise @c FALSE
+ * to allow other handlers to be run. This allows for multiplexing keybindings on the same keys,
+ * depending on the focused widget (or context). If the callback is NULL the group's handler will
+ * be invoked, but the same rule applies.
+ *
+ * @param group Group.
+ * @param key_id Keybinding index for the group.
+ * @param handler Function to call when activated, or @c NULL to use the group callback.
+ * @param user_data User data passed back to the handler.
+ * @param key (Lower case) default key, e.g. @c GDK_j, but usually 0 for unset.
+ * @param mod Default modifier, e.g. @c GDK_CONTROL_MASK, but usually 0 for unset.
+ * @param kf_name Key name for the configuration file, such as @c "menu_new".
+ * @param label Label used in the preferences dialog keybindings tab. May contain
+ * underscores - these won't be displayed.
+ * @param menu_item Optional widget to set an accelerator for, or @c NULL.
+ * @return The keybinding - normally this is ignored.
+ *
+ * @since 1.25
+ * @see See plugin_set_key_group_with_handler
+ **/
+GeanyKeyBinding *keybindings_add_item_with_handler(GeanyKeyGroup *group, gsize key_id,
```

Comment:
just noticed it's "*add*_item", which is different from what we already have, and a bit misleading as it actually doesn't add anything -- it sets it